### PR TITLE
Include total coinhours in /coinSupply

### PR DESCRIPTION
**Pull request by therealssj**
Resolves: #865

### CODE REVIEW

File: src/visor/readable.go [hunk 1]
```diff
@@ -127,6 +127,8 @@ type ReadableTransactionOutput struct {
 type ReadableTransactionInput struct {
 	Hash    string `json:"uxid"`
 	Address string `json:"owner"`
+	Coins   string `json:"coins"`
+	Hours   uint64 `json:"hours"`
```
Comment (gz-c) [inline src/visor/readable.go]: hours should always be strings

File: src/gui/explorer.go [hunk 2]
```diff
@@ -123,10 +125,18 @@ func coinSupply(gateway *daemon.Gateway, w http.ResponseWriter, r *http.Request)
 		return nil, nil
 	}
 
+	totalCoinHours, err := gateway.GetTotalCoinhours()
```
Comment (gz-c) [inline src/gui/explorer.go]: Should call GetUnspentOutputs() directly and reuse the result to calculate total coin hours and also filter the unlocked addresses which is done in the lines above

File: src/visor/readable.go (hunk 1 above)
Comment (therealssj) [reply src/visor/readable.go]: aren't hours like elapsed time?
they are stored as `uint64` everywhere else

File: src/gui/explorer.go (hunk 2 above)
Comment (therealssj) [reply src/gui/explorer.go]: when u say filter, do u mean skip the unlocked addresses?

File: src/visor/readable.go (hunk 1 above)
Comment (gz-c) [reply src/visor/readable.go]: API policy going forward is to using strings for hours

**Comment by Senyoret1** [comment]
Please, see https://github.com/skycoin/skycoin/issues/867#issuecomment-359140328

File: src/gui/explorer.go [hunk 3]
```diff
@@ -155,21 +152,66 @@ func getTransactionsForAddress(gateway *daemon.Gateway) http.HandlerFunc {
 			return
 		}
 
+		// Get start and end pagination index
+		start := r.FormValue("start")
```
Comment (gz-c) [inline src/gui/explorer.go]: would be better written as 

```go
startIndex := 0
start := r.FormValue("start")
if start != "" {
    // atoi, check err, check startIndex >= 0
}
```

File: src/gui/explorer.go [hunk 4]
```diff
@@ -155,21 +152,66 @@ func getTransactionsForAddress(gateway *daemon.Gateway) http.HandlerFunc {
...
+		}
+
 		cipherAddr, err := cipher.DecodeBase58Address(addr)
 		if err != nil {
 			wh.Error400(w, "invalid address")
 			return
 		}
 
 		txns, err := gateway.GetAddressTxns(cipherAddr)
+		totalLen := len(txns.Txns)
```
Comment (gz-c) [inline src/gui/explorer.go]: This reduces the amount of data sent in the request but the real bottleneck is the db query, is it possible to improve query performance with the range parameters?

File: src/visor/readable.go [hunk 5]
```diff
@@ -145,10 +147,12 @@ func NewReadableTransactionOutput(t *coin.TransactionOutput, txid cipher.SHA256)
 }
 
 // NewReadableTransactionInput creates readable transaction input
-func NewReadableTransactionInput(uxID string, ownerAddress string) ReadableTransactionInput {
+func NewReadableTransactionInput(uxID string, ownerAddress string, coins string, hours string) ReadableTransactionInput {
```
Comment (gz-c) [inline src/visor/readable.go]: This function signature can be rewritten as `(uxID, ownerAddress, coins, hours string)`

**Comment by gz-c** [comment]
The pagination changes should be a separate PR from showing the total coin hours (which would have been merged if not for the pagination changes which may take a while)

**Comment by therealssj** [comment]
@gz-c I have removed the pagination changes

**Comment by therealssj** [comment]
does this need to go into the changelog?
otherwise looks good for the merge

File: src/visor/readable.go [hunk 6]
```diff
@@ -127,6 +127,8 @@ type ReadableTransactionOutput struct {
 type ReadableTransactionInput struct {
 	Hash    string `json:"uxid"`
 	Address string `json:"owner"`
+	Coins   string `json:"coins"`
```
Comment (gz-c) [inline src/visor/readable.go]: this is not related to coinhour_supply changes

**Comment by gz-c** [comment]
Remove the changes for #866 too, one PR per issue

**Comment by therealssj** [comment]
removed the other changes and updated changelog and readme

File: src/gui/explorer.go [hunk 7]
```diff
@@ -42,6 +31,8 @@ type CoinSupply struct {
 	TotalSupply string `json:"total_supply"`
 	// MaxSupply is the maximum number of coins to be distributed ever
 	MaxSupply string `json:"max_supply"`
+	// CoinHourSupply is the total number of coinhours in the distribution addresses
+	CoinHourSupply string `json:"coinhour_supply"`
```
Comment (gz-c) [inline src/gui/explorer.go]: Can we do `current_coinhour_supply` (exclude all distribution addresses) and `total_coinhour_supply` (same as `coinhour_supply` is here) to match `current_supply` and `total_supply`?

File: src/gui/explorer.go [hunk 8]
```diff
@@ -104,31 +106,43 @@ func coinSupply(gateway *daemon.Gateway, w http.ResponseWriter, r *http.Request)
...
+
+	// Get coin hours from confirmed unspents only
 	var totalCoinHours uint64
 	for _, out := range lockedOutputs.HeadOutputs {
 		totalCoinHours += out.Hours
 	}
 
+	// get current coin hours which excludes all distribution addresses
+	coinHourFilter = append(coinHourFilter, daemon.FbyAddressesNotIncluded(unlockedAddrs))
+	normalOutputs, err := gateway.GetUnspentOutputs(coinHourFilter...)
```
Comment (gz-c) [inline src/gui/explorer.go]: Is it not possible to apply the filters ourselves? It's querying the full database 3 times in this endpoint now

**Comment by gz-c** [comment]
need to do `make format`, one of the files is not formatted
